### DISCUSSION TITLE
有支持REALITY配置的一键脚本吗

### DISCUSSION THREAD

**Discussion by GreatMichaelLee** [XTLS/Xray-core#1770] [Q&A]
如题。

**Comment by zxcvos** [comment]
[Xray-REALITY 管理脚本](https://github.com/zxcvos/Xray-script)

你看看符不符合你要求

**Reply by GreatMichaelLee** [reply]
> [Xray-REALITY 管理脚本](https://github.com/zxcvos/Xray-script)
> 
> 你看看符不符合你要求
--2023-03-20 17:40:44--  https://raw.githubusercontent.com/zxcvos/Xray-script/main/reality.sh
Resolving raw.githubusercontent.com (raw.githubusercontent.com)... 2606:50c0:8002::154, 2606:50c0:8001::154, 2606:50c0:8000::154, ...
Connecting to raw.githubusercontent.com (raw.githubusercontent.com)|2606:50c0:8002::154|:443... connected.
HTTP request sent, awaiting response... 200 OK
Length: 26168 (26K) [text/plain]
Saving to: ‘/root/Xray-script.sh’

/root/Xray-script.sh                              100%[==========================================================================================================>]  25.55K  --.-KB/s    in 0s

2023-03-20 17:40:44 (67.4 MB/s) - ‘/root/Xray-script.sh’ saved [26168/26168]

root@Vultr-LAX:~# bash ${HOME}/Xray-script.sh
/root/Xray-script.sh: line 12: $'\r': command not found
/root/Xray-script.sh: line 17: $'\r': command not found
': not a valid identifiere 18: declare: `domain
': not a valid identifiere 19: declare: `domain_path
': not a valid identifiere 20: declare: `new_port
/root/Xray-script.sh: line 21: $'\r': command not found
/root/Xray-script.sh: line 22: syntax error near unexpected token `$'{\r''
'root/Xray-script.sh: line 22: `function _info() {

**Reply by zxcvos** [reply]
~~你这什么系统？怎么空行会 not found :P~~ 是我的锅，我自己测试的时候都是cv到服务器所以没毛病，但是在丢到仓库的时候没注意到换行符:(

**Reply by GreatMichaelLee** [reply]
> ~你这什么系统？怎么空行会 not found :P~ 是我的锅，我自己测试的时候都是cv到服务器所以没毛病，但是在丢到仓库的时候没注意到换行符:(

这脚本挺好的，感谢，但能否开放一个交互配置，让我可以制定uuid,因为我要配置负载均衡，必须相同的uuid,不然每次你脚本随机生成我每个vps里还要去手动改配置文件，辛苦哈

**Reply by imtv** [reply]
配置文件里改一下UUID并不会比你所说的交互配置填入复杂多少，真的。
nano /usr/local/etc/xray/config.json
进去编辑，然后重启一下xray
systemctl restart xray
就好了

**Reply by zxcvos** [reply]
> > ~你这什么系统？怎么空行会 not found :P~ 是我的锅，我自己测试的时候都是cv到服务器所以没毛病，但是在丢到仓库的时候没注意到换行符:(
> 
> 这脚本挺好的，感谢，但能否开放一个交互配置，让我可以制定uuid,因为我要配置负载均衡，必须相同的uuid,不然每次你脚本随机生成我每个vps里还要去手动改配置文件，辛苦哈

加了加了:D

**Reply by crazypeace** [reply]
> > ~你这什么系统？怎么空行会 not found :P~ 是我的锅，我自己测试的时候都是cv到服务器所以没毛病，但是在丢到仓库的时候没注意到换行符:(
> 
> 这脚本挺好的，感谢，但能否开放一个交互配置，让我可以制定uuid,因为我要配置负载均衡，必须相同的uuid,不然每次你脚本随机生成我每个vps里还要去手动改配置文件，辛苦哈

你要不要试试这个脚本, 可以把UUID当参数传给脚本, 这样你只要把命令行保存好, 比如存成快捷按钮, 这样每次搭出来的节点, 参数都是一样的.
https://github.com/crazypeace/xray-vless-reality

**Comment by mmrabbani** [comment]
Here you are.
https://github.com/sajjaddg/xray-reality

**Comment by jinwyp** [comment]
最全的一键脚本 没有之一 
https://github.com/jinwyp/one_click_script/blob/master/README2_CN.md

**Comment by abcjeff** [comment]
脚本一大堆，vps一大堆，最头疼是申请有效的域名

**Reply by jinwyp** [reply]
> 脚本一大堆，vps一大堆，最头疼是申请有效的域名

所以 Reality 可以不需要域名安装 直接用脚本 选择18 安装 非常方便
https://github.com/jinwyp/one_click_script/blob/master/README2_CN.md

**Comment by crazypeace** [comment]
不喜欢脚本菜单太长的朋友， 直接用这个极简一键脚本
https://github.com/crazypeace/xray-vless-reality
